components/org.wso2.carbon.identity.api.user.totp/org.wso2.carbon.identity.api.user.totp.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/totp/v1/core/TOTPService.java

```diff
@@ -292,4 +314,23 @@ private APIError handleError(Response.Status status, TOTPConstants.ErrorMessage
         return new ErrorResponse.Builder().withCode(errorEnum.getCode()).withMessage(errorEnum.getMessage())
                 .withDescription(errorEnum.getDescription());
     }
+
+    private boolean isValidAuthenticationType() {
+
+        /*
+        Check whether the request is authenticated with basic auth. TOTP endpoint should not be allowed for basic
+        authentication. This approach can be improved by providing a Level of Assurance (LOA) and checking that in
+        TOTPService.
+         */
+        if (IdentityUtil.threadLocalProperties.get().get(TOTPConstants.AUTHENTICATED_WITH_BASIC_AUTH) != null
```

Comment:
`IdentityUtil.threadLocalProperties.get().get(TOTPConstants.AUTHENTICATED_WITH_BASIC_AUTH) != null` is this needed? Wouldn't parseBoolean return false if the property is null?

Reply:
Resolved